### PR TITLE
Updating Alarm syscall driver to 2.0 syscall API

### PR DESCRIPTION
**Pull request by phil-levis**
### Pull Request Overview

This pull requests updates the alarm system call driver to use the 2.0 syscall API.

### Testing Strategy

This pull request was tested by running two copies of multi_alarm_test in parallel on the imix platform. I found a few bugs while doing so, which led me to update libtock-c's alarm code to use the new (reference, dt) system calls. It had still be using just an absolute time, which if there are long delays due to printfs can cause the hang-on-alarm-in-the-past bug.

### Documentation Updated

- [x] Updated the relevant files in `/docs`, or no updates are required.

### Formatting

- [x] Ran `make prepush`.

### CODE REVIEW

File: capsules/src/alarm.rs [hunk 1]
```diff
@@ -227,6 +242,7 @@ impl<'a, A: Alarm<'a>> LegacyDriver for AlarmDriver<'a, A> {
                         let future_time = data;
                         let dt = future_time.wrapping_sub(reference);
                         // if previously unarmed, but now will become armed
+                        debug!("Alarm.rs: setting to {} + {}", reference, dt);
```
Comment (hudson-ayers) [inline capsules/src/alarm.rs:245]:
```suggestion
```

**Review by hudson-ayers — APPROVED** [review]
Looks good except for the leftover debug statement

File: capsules/src/alarm.rs (hunk 1 above)
Comment (arjundeopujari) [reply capsules/src/alarm.rs:245]: Yes. Looks good.

**Pull request merged**